onert-micro/eval-driver/TrainingDriver.cpp

```diff
@@ -100,117 +92,236 @@ int entry(int argc, char **argv)
+  const char *input_target_test_data_path = nullptr;
+  int32_t num_train_data_samples = 0;
+  int32_t num_test_data_samples = 0;
+
+  if (argc == 10)
+  {
+    circle_model_path = argv[1];
+    wof_file_path = argv[2];
+    output_trained_file_path = argv[3];
+    input_input_train_data_path = argv[4];
+    input_target_train_data_path = argv[5];
+    input_input_test_data_path = argv[6];
+    input_target_test_data_path = argv[7];
+    num_train_data_samples = atoi(argv[8]);
+    num_test_data_samples = atoi(argv[9]);
+  }
+  else if (argc == 9)
+  {
+    circle_model_path = argv[1];
+    output_trained_file_path = argv[2];
+    input_input_train_data_path = argv[3];
+    input_target_train_data_path = argv[4];
+    input_input_test_data_path = argv[5];
+    input_target_test_data_path = argv[6];
+    num_train_data_samples = atoi(argv[7]);
+    num_test_data_samples = atoi(argv[8]);
+  }
+  else
+  {
+    throw std::runtime_error("Unknown commands number\n");
+  }
 
-  printf("Run train dataset:\n");
-  for (int i = 0; i < num_train_data_samples; ++i)
+  DataBuffer circle_model = readFile(circle_model_path);
+  DataBuffer wof_data;
+  // If defined wof file
+  if (wof_file_path != nullptr)
+    wof_data = readFile(wof_file_path);
+
+  // Configure training mode
+  onert_micro::OMConfig config;
+  // If defined wof file
+  if (wof_file_path != nullptr)
+    config.wof_ptr = nullptr;
+
+  // Set user defined training settings
+  const uint32_t training_epochs = 50;
+  const float lambda = 0.01f;
+  const uint32_t BATCH_SIZE = 64;
+  const uint32_t INPUT_SIZE = 13;
+  const uint32_t OUTPUT_SIZE = 1;
+  const uint32_t num_train_layers = 0;
+  const onert_micro::OMLoss loss = onert_micro::MSE;
+  const onert_micro::OMTrainOptimizer train_optim = onert_micro::ADAM;
+  const float beta = 0.9;
+  const float beta_squares = 0.999;
+  const float epsilon = 1e-07;
+
+  config.train_mode = true;
   {
-    auto input_data = reinterpret_cast<char *>(interpreter.allocateInputTensor(0));
+    onert_micro::OMTrainingContext train_context;
+    train_context.batch_size = BATCH_SIZE;
+    train_context.num_of_train_layers = num_train_layers;
+    train_context.lambda = lambda;
+    train_context.loss = loss;
+    train_context.optimizer = train_optim;
+    train_context.beta = beta;
+    train_context.beta_squares = beta_squares;
+    train_context.epsilon = epsilon;
+
+    config.training_context = train_context;
+  }
 
-    std::memcpy(input_data, train_data_u8, size);
+  // Create training interpreter and import models
+  onert_micro::OMTrainingInterpreter train_interpreter;
+  train_interpreter.importTrainModel(circle_model.data(), config);
 
-    interpreter.interpret();
-    auto data = reinterpret_cast<float *>(interpreter.readOutputTensor(0));
+  // Temporary buffer to read input data from file using BATCH_SIZE
+  float training_input[BATCH_SIZE * INPUT_SIZE];
+  float training_target[BATCH_SIZE * OUTPUT_SIZE];
```

Comment:
label data for CCE is one hot encoding. You will feed int label data by converting into float ?

Reply:
Yes, I think it is one of the preprocessing step. And we have (for cross-etropy task) as input already preprocessed  one-hot encoded float data. Do you think it is better to move it into runtime?